### ISSUE TITLE
TimeUnit Transforms: Different behavior with mark_bar and mark_boxplot

### ISSUE THREAD

**Issue by suddenlyfleck** [altair-viz/altair#2423]
Consider these two plots:

```
source = data.github()

barPlot = alt.Chart(source).mark_bar().encode(
    x='yearmonth(time):O',
    y='count:Q'
)

boxPlot = alt.Chart(source).mark_boxplot().encode(
    x='yearmonth(time):O',
    y='count:Q',
)

barPlot | boxPlot
```
![visualization](https://user-images.githubusercontent.com/58106451/109331019-9d851100-785c-11eb-80f8-5c62acc4a161.png)

Two things seem unintended to me:

1) the x-axis labels for the box plot, after a TimeUnit Transform `yearmonth(time)`, are too verbose.

2) I don't understand the `null` column.

The Vega Editor does these things differently. However, I like the label arrangement of altair's mark_bar() better.

Hope this is useful for you to improve altair.

And last but not least: Thank you for this amazing plotting library! I am blown away. Best thing since sliced bread, no shit. Thank you very much! Altair is incredible!

**Comment by jakevdp** [comment]
Thanks! This looks like a Vega-Lite bug that has been fixed in more recent versions (this is why it works in the Vega Editor). When we update Altair to a new Vega-Lite, the issue will be fixed.

By the way, the reason the bars look strange in the Vega Editor is that there are multiple entries per month; you can fix this by aggregating the count (`y='sum(count)'`).

**Comment by suddenlyfleck** [comment]
I looked at Vega Editor's output using `y='sum(count)'`. I updated the line to make the two examples more similar before posting here.

Anyway, my comment

> I like the label arrangement of altair's mark_bar() better.

was coming when already using `y='sum(count)'`.

Thanks for the quick response. I hope I will think of closing this issue, once I see it resolved with a future version.

Keep up the amazing work. I'm looking hopefully into the future here... :D
